crates/ra_parser/src/grammar/expressions/atom.rs

```diff
@@ -416,6 +416,7 @@ pub(crate) fn match_arm_list(p: &mut Parser) {
 //         | X => (),
 //         box X => (),
 //         Some(box X) => (),
+//         Some(box Test{field: 0}) => (),
```

Comment:
Hm, actually, I think these last three lines are misplaced. I specifically looked for this in the previous PR and somehow still missed it =/

The purpose of these inline tests is to show what exactly each specific bit of code parses. box patterns are *not* handled in this function.

I see that you've added tests to `fn box_pat`, where they belong, so these last three lines should be removed.